src/start_server.ts

```diff
@@ -81,44 +81,171 @@ async function applyDefaultOptions(options: ServerOptions):
         certPath: options.certPath || 'cert.pem',
         keyPath: options.keyPath || 'key.pem',
       });
+      withDefaults.packageName = options.packageName ||
+          bowerConfig(withDefaults.root).name || path.basename(process.cwd());
       return withDefaults;
     }
 
-/**
- * If port unspecified/negative, finds an open port on localhost
- * @param {number} port
- * @returns {Promise<number>} Promise of open port
- */
-async function nextOpenPort(port: number):
-    Promise<number> {
-      if (!port || port < 0) {
-        port = await new Promise<number>(resolve => {
-          // TODO: Switch from `find-port` to `get-port`. `find-port` always
-          // resolves a port number even if none are available. The error-event
-          // handler in `startWithPort` catches the issue.
-          findPort(8080, 8180, (ports: number[]) => {
-            resolve(ports[0]);
-          });
-        });
-      }
-      return port;
-    }
 
 /**
  * @return {Promise} A Promise that completes when the server has started.
+ * @deprecated Please use `startServers` instead. This function will be removed
+ *     in a future release.
  */
 export async function startServer(options: ServerOptions):
     Promise<http.Server> {
-      options = options || {};
-      assertNodeVersion(options);
-      try {
-        return await startWithPort(options)
-      } catch (e) {
-        console.error('ERROR: Server failed to start:', e);
-        throw new Error(e);
-      }
+      return (await _startServer(options)).server;
     }
 
+async function _startServer(options: ServerOptions) {
+  options = options || {};
+  assertNodeVersion(options);
+  try {
+    const fullOptions = await applyDefaultOptions(options);
+    const app = getApp(options);
+    const server = await startWithPort(fullOptions, app);
+    return {app, server};
+  } catch (e) {
+    console.error('ERROR: Server failed to start:', e);
+    throw new Error(e);
+  }
+}
+
+export type ServerInfo = MainlineServer | VariantServer | DispatchServer;
+
+/**
+ * The `default` or `primary` server. If only one ServerInfo is returned from
+ * startServers it must be a MainlineServer. This is the server that's running
+ * with the default configuration and not running a variant configuration.
+ */
+export interface MainlineServer {
```

Comment:
It's not yet an AppServer is it? All of the VariantServers also serve out the application directory I think